### ISSUE TITLE
Cannot create Wiki pages. Backward compatibility broken (git -c flag not in version 1.7.1)

### ISSUE THREAD

**Issue by jhit** [gogits/gogs#3833]
- Gogs version (or commit ref): 0.9.99.0903
- Git version:  1.7.1
- Operating system:  CentOS 6
- Database (use `[x]`): 
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

gogs.log
```
2016/10/31 14:31:58 [...routers/repo/wiki.go:216 NewWikiPost()] [E] AddWikiPage: CommitChanges: exit status 129 - Unknown option: -c
usage: git [--version] [--exec-path[=GIT_EXEC_PATH]] [--html-path]
           [-p|--paginate|--no-pager] [--no-replace-objects]
           [--bare] [--git-dir=GIT_DIR] [--work-tree=GIT_WORK_TREE]
           [--help] COMMAND [ARGS]
```

## Description
Any creation of a wiki page fails with Error 500, even in freshly created repositories. After the error occured the first time the wiki function always fails with error 500 for this repository.

As the error message clearly states gogs tries to use the commandline flag `-c` which is not available in git version `1.7.1`. Nevertheless the [docs](https://gogs.io/docs/installation) state that the minimum version for git is `1.7.1`.
...

**Comment by jhit** [comment]
For anyone experiencing this issue:
If you updated your git version like I did and still cannot open the wiki pages, here is what I did to get the wiki working again.
 1. Find the folder where you store your gogs repositories. It is shown as the **Repository Root Path** on the `Admin Panel -> Configuration` page
 2. Search for the `*.wiki.git` folders in that location
 3. Delete the `my_repo_with_defect.wiki.git`. Attention! Danger Zone! Do not delete the main repositories, that are the ones **without** `*.wiki.*` in their folder name.
